wd_demo.au3

```diff
@@ -175,37 +175,39 @@ Func RunDemo($idDebugging, $idBrowsers, $idUpdate, $idHeadless)
 
 	ConsoleWrite("> _WD_Startup" & @CRLF)
 	Local $iWebDriver_PID = _WD_Startup()
-	Local $iError = @error
-
-	If $iError = $_WD_ERROR_Success Then
-		ConsoleWrite("> _WD_CreateSession" & @CRLF)
-		$sSession = _WD_CreateSession($sCapabilities)
-		$iError = @error
-
-		If $iError = $_WD_ERROR_Success Then
-			For $iIndex = 0 To UBound($aDemoSuite, $UBOUND_ROWS) - 1
-				If $aDemoSuite[$iIndex][1] Then
-					ConsoleWrite("+ Running: " & $aDemoSuite[$iIndex][0] & @CRLF)
-					If $aDemoSuite[$iIndex][2] Then
-						Call($aDemoSuite[$iIndex][0], $sBrowser)
-					Else
-						Call($aDemoSuite[$iIndex][0])
-					EndIf
-					$iError = @error
-					If $iError = $_WD_ERROR_UserAbort Then
-						ConsoleWrite("- Aborted: " & $aDemoSuite[$iIndex][0] & @CRLF)
-						ExitLoop
-					EndIf
-					ConsoleWrite("+ Finished: " & $aDemoSuite[$iIndex][0] & @CRLF)
-				Else
-					ConsoleWrite("> Bypass: " & $aDemoSuite[$iIndex][0] & @CRLF)
-				EndIf
-			Next
-
+	If @error And _RunDemo_errorHandling(@error, @extended, $iWebDriver_PID, $sSession) Then Return
+
+	ConsoleWrite("> _WD_CreateSession" & @CRLF)
+	$sSession = _WD_CreateSession($sCapabilities)
+	If @error And _RunDemo_errorHandling(@error, @extended, $iWebDriver_PID, $sSession) Then Return
+
+	Local $iError
+	For $iIndex = 0 To UBound($aDemoSuite, $UBOUND_ROWS) - 1
+		If $aDemoSuite[$iIndex][1] Then
```

Comment:
@mlipok 
I still recommend to get rid of unnecessary nesting by `ContinueLoop` 😉 .

``` autoit
	Local $iError
	For $iIndex = 0 To UBound($aDemoSuite, $UBOUND_ROWS) - 1
		If Not $aDemoSuite[$iIndex][1] Then
			ConsoleWrite("> Bypass: " & $aDemoSuite[$iIndex][0] & @CRLF)
			ContinueLoop
		EndIf
	
		ConsoleWrite("+ Running: " & $aDemoSuite[$iIndex][0] & @CRLF)
		If $aDemoSuite[$iIndex][2] Then
			Call($aDemoSuite[$iIndex][0], $sBrowser)
		Else
			Call($aDemoSuite[$iIndex][0])
		EndIf
		$iError = @error
		If $iError = $_WD_ERROR_UserAbort Then
			ConsoleWrite("- Aborted: " & $aDemoSuite[$iIndex][0] & @CRLF)
			ExitLoop
		ElseIf $iError Then
			ConsoleWrite("! Error: " & $aDemoSuite[$iIndex][0] & @CRLF)
			ExitLoop
		EndIf
		ConsoleWrite("+ Finished: " & $aDemoSuite[$iIndex][0] & @CRLF)
	Next
```

Reply:
https://github.com/Danp2/WebDriver/issues/179